### ISSUE TITLE
fatal error: module 'mmkv' not found

### ISSUE THREAD

**Issue by guodongyangwen** [Tencent/MMKV#669]
在使用Android Studio运行flutter项目时，出现这个问题，我看官网说是执行pod install然后通过脚本把mmkv改成mmkvflutter，但是通过Android Studio运行第一次是会执行pod install，但是后续运行程序就不会执行pod install,导致后续还是会改成mmkv，导致找不到这个模块

下面是我的日志：

Automatically signing iOS for device deployment using specified development team in Xcode project: YNDE2492ZY
Warning: Podfile is out of date
  This can cause a mismatched version of Flutter to be embedded in your app, which may result in App Store submission rejection or crashes.
  If you have local Podfile edits you would like to keep, see https://github.com/flutter/flutter/issues/24641 for instructions.
To regenerate the Podfile, run:
  rm ios/Podfile

Running Xcode build...
Xcode build done.                                           24.0s
Failed to build iOS app
Error output from Xcode build:
↳
    2021-04-21 14:19:56.950 xcodebuild[91403:1792294]  DTDeviceKit: deviceType from 00008020-000571843484002E was NULL
    2021-04-21 14:19:57.112 xcodebuild[91403:1792299]  DTDeviceKit: deviceType from 00008020-000571843484002E was NULL
    2021-04-21 14:19:57.112 xcodebuild[91403:1792299]  DTDeviceKit: deviceType from 00008020-000571843484002E was NULL
    ** BUILD FAILED **

Xcode's output:
↳
    In file included from /Users/admin/Documents/Flutter_App_Toca/ios/Runner/AppDelegate.m:1:
    /Users/admin/Documents/Flutter_App_Toca/ios/Runner/AppDelegate.h:10:32: warning: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Wnullability-completeness]
    - (void)tcConfigOpenUrl:(NSURL *)url;
                                   ^
    /Users/admin/Documents/Flutter_App_Toca/ios/Runner/AppDelegate.h:10:32: note: insert '_Nullable' if the pointer may be null
    - (void)tcConfigOpenUrl:(NSURL *)url;
                                   ^
                                    _Nullable
    /Users/admin/Documents/Flutter_App_Toca/ios/Runner/AppDelegate.h:10:32: note: insert '_Nonnull' if the pointer should never be null
    - (void)tcConfigOpenUrl:(NSURL *)url;
                                   ^
                                    _Nonnull
    1 warning generated.
    /Users/admin/Documents/Flutter_App_Toca/ios/Runner/GeneratedPluginRegistrant.m:28:9: fatal error: module 'mmkv' not found
    @import mmkv;
     ~~~~~~~^~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

**Comment by lingol** [comment]
I recommend using Xcode to run flutter App on iOS device.

**Comment by vladaman** [comment]
Having same issue:

### The version of MMKV
v1.2.8

### The platform of MMKV
iOS
```
flutter doctor -v
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.7 19H524 darwin-x64, locale cs)
    • Flutter version 1.22.5 at /Users/user/development/flutter
    • Framework revision 7891006299 (5 months ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, set ANDROID_SDK_ROOT to that location.
      You may also want to add it to your PATH environment variable.
 
[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] Connected device (1 available)            
    • iPhone (mobile) • xxxxxxxx
```

### The installation of MMKV
Cocoapods

### What's the issue?
Unable to run project due to an error:
```
3 warnings generated.
    1 warning generated.
    1 warning generated.
    /Users/user/development/myapp/ios/Runner/GeneratedPluginRegistrant.m:100:9: fatal error: module 'mmkv' not found
    @import mmkv;
     ~~~~~~~^~~~
    1 error generated.
```

Steps done in order to integrate the project (as described in manual)

1. Added to pubspec.yaml
```yaml
dependencies:
  mmkv: ">=1.2.8"
```
2. Added fix_mmkv_plugin_name function inti ios/Podfile
3. `cd ios/ && pod install`
4. `flutter packages get && flutter run`

@lingol unfortunately running from xcode I get same error.

**Comment by lingol** [comment]
Duplicated. You should search issues before making a new issue.

**Comment by lingol** [comment]
And you should run pod install after each time you run flutter get.

**Comment by vladaman** [comment]
I figured where the problem is. In docs you mentioned:

 > _For a purely flutter App, add this function fix_mmkv_plugin_name() to ios/Podfile, invoke it **before** calling any flutter_xxx() functions. Run pod install and we are all set._

The correct solution is to paste the code **between flutter_root** and any **flutter_xxx**()* . Placing the code before flutter_root causes compiling errors issue.

## Solution: ##

 Move code snipped after **flutter_root** function

**Comment by lingol** [comment]
Thanks for your finding. I will check it out after the holiday.

**Comment by lingol** [comment]
@vladaman What do you mean by saying `Move code snipped after flutter_root function`? I can not reproduce this solution on my side. 
Can you provide a working demo?

**Comment by lingol** [comment]
According to [the official document of Flutter](https://flutter.dev/docs/development/add-to-app/ios/project-setup#option-a---embed-with-cocoapods-and-the-flutter-sdk), `pod install` is required after each time you run `flutter pub get`.

Looks like there isn't any elegant solution here.

**Comment by trimaximus** [comment]
Having same issue. 
@lingol u mentioned that 
>`pod install`  is required after each time you run `flutter pub get`.

I found that launching `lib/main.dart` with `pod install` will generate `GeneratedPluginRegistrant.m`:
```Objective-C
#if __has_include(<mmkvflutter/MMKVPlugin.h>)
#import <mmkvflutter/MMKVPlugin.h>
#else
@import mmkvflutter;
#endif
``` 
But without `pod install`:
```Objective-C
#if __has_include(<mmkv/MMKVPlugin.h>)
#import <mmkv/MMKVPlugin.h>
#else
@import mmkv;
#endif
``` 

This way will definitely lead to `mmkv` module not fount.

Each time launching runs `pod install` is inelegant, here comes my solution: adding `run script` in `Xcode`.
```shell
sed -i "" "s/<mmkv\/MMKVPlugin.h>/<mmkvflutter\/MMKVPlugin.h>/g" ${SRCROOT}/Runner/GeneratedPluginRegistrant.m
sed -i "" "s/@import mmkv;/@import mmkvflutter;/g" ${SRCROOT}/Runner/GeneratedPluginRegistrant.m
/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build
```
That works fine for me.

**Comment by lingol** [comment]
@trimaximus 
You are some pro users of MMKV that I rarely meet. Thanks for your contribution.
